src/llm/services/llm.service.ts

```diff
@@ -0,0 +1,493 @@
+              typeof part === 'object' &&
+              'type' in part &&
+              part.type === 'text' &&
+              'text' in part &&
+              typeof part.text === 'string'
+            ) {
+              return {
+                ...part,
+                text: this.sanitizeMessageContent(part.text),
+              }
+            }
+            return part
+          }),
+        } as ChatCompletionMessageParam
+      }
+      return message
+    })
+  }
+
+  /**
+   * Extracts content and tool calls from a chat completion response.
+   * Returns both content and tool calls separately for unambiguous handling.
+   * Utility function for response processing.
+   */
+  extractCompletionContent(completion: ChatCompletion): {
+    content: string
+    toolCalls?: ToolCall[]
+  } {
+    const message = completion.choices[0]?.message
+
+    if (!message) {
+      return { content: '' }
+    }
+
+    if (message.tool_calls && message.tool_calls.length > 0) {
+      const toolCalls: ToolCall[] = message.tool_calls.map((toolCall) => ({
+        id: toolCall.id,
+        type: toolCall.type,
+        function: {
+          name: toolCall.function?.name || '',
+          arguments: toolCall.function?.arguments || '',
+        },
+      }))
+      return {
+        content: message.content || '',
+        toolCalls,
+      }
+    }
+
+    return { content: message.content || '' }
+  }
+
+  /**
+   * Internal method to make a chat completion API call.
+   * Handles the actual TogetherAI API request with proper configuration.
+   */
+  private async callChatCompletion({
+    model,
+    messages,
+    temperature,
+    topP,
+    maxTokens,
+    timeout,
+    userId,
+  }: {
+    model: string
+    messages: ChatCompletionMessageParam[]
+    temperature: number
+    topP: number
+    maxTokens?: number
+    timeout: number
+    userId?: string
+  }): Promise<LlmCompletionResult> {
+    const sanitizedMessages = this.sanitizeMessages(messages)
+    const userIdentification = this.prepareUserIdentification(userId)
+
+    const requestParams: Parameters<
+      typeof this.client.chat.completions.create
+    >[0] = {
+      model,
+      messages: sanitizedMessages,
+      temperature,
+      top_p: topP,
+      ...(maxTokens && { max_tokens: maxTokens }),
+      ...userIdentification,
+      stream: false,
+    }
+
+    this.logger.debug('Making TogetherAI API request', {
+      model,
+      baseURL: this.client.baseURL,
+      messageCount: sanitizedMessages.length,
+      hasUserId: !!userId,
+    })
+
+    try {
+      const completion = (await this.client.chat.completions.create(
+        requestParams,
+        {
+          timeout,
+        },
+      )) as ChatCompletion
+
+      const { content, toolCalls } = this.extractCompletionContent(completion)
+      const tokens = completion.usage?.total_tokens || 0
+
+      return {
+        content: content.trim(),
+        tokens,
+        ...(toolCalls && { toolCalls }),
+      }
```

Comment:
Can we remove the `toolCalls`-related code here for now, since we aren't using tools yet?